pkg/apis/datavolumecontroller/v1alpha1/types.go

```diff
@@ -53,6 +54,11 @@ type DataVolumeSourcePVC struct {
 	Name      string `json:"name,omitempty"`
 }
 
+// DataVolumeSourceUpload provides the parameters to create a Data Volume by uploading the source
+type DataVolumeSourceUpload struct {
+	ShouldUpload bool `json:"shouldUpload,omitempty"`
```

Comment:
Fine with me. However, the controller needs to check existence of this field "target", and the check should be something like if 'dataVolume.Spec.Source.target != "" '.  Is it ok to set it now to 'default' since the controller ignore it anyway. Otherwise I need to check if 'dataVolume.Spec.Source.target == "" ' since it is a string. and we will have to change this in the future versus changing the yaml file

Reply:
I will set the yaml to be
target: "local" for now

Reply:
I don't understand this conversation.  It seems like a bad idea to pass around something arbitrary.  Shouldn't the presence of `upload: {}` in the source field of the spec be enough to indicate that an upload is requested?  If some time in the future we have one or more annotations to pass wrt to upload we can add them at that time.

Reply:
done